### ISSUE TITLE
'createUnionType' ignores extends

### ISSUE THREAD

**Issue by WimSuenensDIRACIndustries** [MichalLytek/type-graphql#587]
When you use 'extends' on the ObjectType, createUnionType only resolves to the base ObjectType and ignores the extended objects.
This also applies, when you wouldn't use 'extends', but define the properties on the other objects. As soon as the new object defines the same properties (+ some additional), it is ignored in the UnionType.

```typescript
import { ObjectType, Field, createUnionType } from "type-graphql";
import { User } from "../../database/postgresql/models/User";

@ObjectType()
export class SuccessMessage {
    @Field()
    success: boolean;

    @Field()
    message: string;
}

@ObjectType()
export class SuccessMessage_Token extends SuccessMessage {
    @Field()
    accessToken: string;

    @Field()
    user: User;
}

@ObjectType()
export class ErrorMessage extends SuccessMessage{
    @Field()
    success: boolean;

    @Field()
    error: string;
}

export const ResponseMessageUnion = createUnionType({
    name: "ResponseMessage", // the name of the GraphQL union
    types: () => [
            SuccessMessage,
            SuccessMessage_Token,
            ErrorMessage
        ], // function that returns array of object types classes
    resolveType: value => {
        if ("error" in value) {
            return ErrorMessage;
        }
        if ("accessToken" in value) {
            return SuccessMessage_Token
        }
        if ("error" !in value && "accessToken" !in value) {
            return SuccessMessage;
        }
        return undefined;
    }
});
```

In this case, it's only possible the resolve to 'SuccessMessage'

```typescript
(alias) createUnionType<(typeof SuccessMessage)[]>(config: UnionTypeConfig<(typeof SuccessMessage)[]>): SuccessMessage
import createUnionType
```

**Comment by MichalLytek** [comment]
Unfortunately, that's limitation of TS structural typings and type inference.

Here is an example how fragile it can be:

![image](https://user-images.githubusercontent.com/10618781/77834181-f88d2000-7142-11ea-85e0-e7251046f933.png)

No compile errors, it doesn't complain about missing property and incorreclty narrow the type then.

But I've discovered that using const assertion can fix your problem:

![Code_99cvnoai1N](https://user-images.githubusercontent.com/10618781/77834187-09d62c80-7143-11ea-8380-1b529a8d632e.png)

But this requires some changes in `type-graphql` code, so please wait for a proper fix 😉 

BTW, you have an error in the code - instead of:
```ts
if ("error" !in value && "accessToken" !in value)
```
it should  be:
```ts
if (!("error" in value) && !("accessToken" in value))
```

**Comment by MichalLytek** [comment]
Closing via 2e5f155 🔒 Released in `v0.18.0-beta.15` 🚀

**Comment by WimSuenensDIRACIndustries** [comment]
Thanks for the quick fix!
